src/main/kotlin/org/rust/cargo/toolchain/RsToolchain.kt

```diff
@@ -10,15 +10,16 @@ import com.intellij.openapi.util.io.FileUtil
 import com.intellij.util.io.exists
 import com.intellij.util.text.SemVer
 import org.rust.cargo.toolchain.tools.Cargo
-import org.rust.cargo.toolchain.tools.Rustc
+import org.rust.ide.sdk.flavors.RsSdkFlavor
+import org.rust.ide.sdk.flavors.suggestHomePaths
+import org.rust.stdext.isExecutable
 import java.io.File
-import java.nio.file.Files
 import java.nio.file.Path
 
 data class RsToolchain(val location: Path) {
     val presentableLocation: String = pathToExecutable(Cargo.NAME).toString()
 
-    fun looksLikeValidToolchain(): Boolean = hasExecutable(Cargo.NAME) && hasExecutable(Rustc.NAME)
+    fun looksLikeValidToolchain(): Boolean = RsSdkFlavor.getFlavor(location) != null
```

Comment:
Looks like it's only one usage of `RsSdkFlavor.getFlavor`. Maybe rename it to `looksLikeValidToolchain` and pass necessary implementation inside it?